azurecompute/src/main/java/org/jclouds/azurecompute/compute/AzureComputeServiceAdapter.java

```diff
@@ -16,39 +16,275 @@
+
+   private final AzureComputeApi api;
+   private final String subscriptionId;
+   private final long operationTimeout;
+   private Predicate<String> operationSucceeded;
+
+   @Inject
+   public AzureComputeServiceAdapter(final AzureComputeApi api, @Named(SUBSCRIPTION_ID) String subscriptionId, @Named(OPERATION_TIMEOUT) long operationTimeout) {
+      this.api = checkNotNull(api, "api");
+      this.subscriptionId = checkNotNull(subscriptionId, "subscriptionId");
+      this.operationTimeout = checkNotNull(operationTimeout, "operationTimeout");
+      this.operationSucceeded = retry(new Predicate<String>() {
+         public boolean apply(String input) {
+            return api.getOperationApi().get(input).status() == Operation.Status.SUCCEEDED;
+         }
+      }, operationTimeout, 5, 5, SECONDS);
+   }
 
    @Override
    public NodeAndInitialCredentials<Deployment> createNodeWithGroupEncodedIntoName(
-         String group, String name, Template template) {
-      // TODO Auto-generated method stub
-      return null;
+           final String group, final String name, Template template) {
+
+      // azure-specific options
+      AzureComputeTemplateOptions templateOptions = template.getOptions().as(AzureComputeTemplateOptions.class);
+      final String virtualNetworkName = templateOptions.getVirtualNetworkName().or(DEFAULT_VIRTUAL_NETWORK_NAME);
+      final String addressSpaceAddressPrefix = templateOptions.getAddressSpaceAddressPrefix().or(DEFAULT_ADDRESS_SPACE_ADDRESS_PREFIX);
+      final String subnetName = templateOptions.getSubnetName().or(DEFAULT_SUBNET_NAME);
+      final Set<String> networkSecurityGroupNames = templateOptions.getGroups().isEmpty() ? Sets.<String>newHashSet() : templateOptions.getGroups();
+      final String subnetAddressPrefix = templateOptions.getSubnetAddressPrefix().or(DEFAULT_SUBNET_ADDRESS_PREFIX);
+      final String storageAccountType = templateOptions.getStorageAccountType().or(DEFAULT_STORAGE_SERVICE_TYPE);
+
+      final String loginUser = templateOptions.getLoginUser() != null ? templateOptions.getLoginUser() : DEFAULT_LOGIN_USER;
+      final String loginPassword = templateOptions.getLoginPassword() != null ? templateOptions.getLoginPassword() : DEFAULT_LOGIN_PASSWORD;
+      final String location = template.getLocation().getId();
+      final int[] inboundPorts = template.getOptions().getInboundPorts();
+
+      String storageAccountName = templateOptions.getStorageAccountName().or(generateStorageServiceName(DEFAULT_STORAGE_ACCOUNT_PREFIX));
+
+      // get or create storage service
+      StorageService storageService = tryFindExistingStorageServiceAccountOrCreate(location, storageAccountName, storageAccountType);
+      String storageServiceAccountName = storageService.serviceName();
+
+      // check existence or create virtual network
+      checkExistingVirtualNetworkNamedOrCreate(virtualNetworkName, location, subnetName, addressSpaceAddressPrefix, subnetAddressPrefix);
```

Comment:
Same comment here. as the templateOptions are shared with all nodes, all "get or create" operations should be done *before* this method is run to avoid race conditions.